### ISSUE TITLE
Alignment controls intermittently dissapear on WP 6.2-beta3

### ISSUE THREAD

**Issue by johnstonphilip** [WordPress/gutenberg#48380]
### Description

When using a custom post type, the alignment controls controls intermittently show or do-not show.

### Step-by-step reproduction instructions

1. Register a custom post type
2. Set the post type's `public` to be `false` in the `register_post_type`
3. Go to the block editor for that post type and add a columns block.
4. Notice the alignment controls are not showing

Sample registration code to use:
```
register_post_type(
		'test',
		array(
			'public'       => false,
			'publicly_queryable' => false,
			'has_archive'  => false,
			'show_ui'      => true,
			'show_in_menu' => false,
			'show_in_rest' => true,
			'supports'     => array(
				'editor',
				'custom-fields',
			),
			'labels'       => array(
				'name'          => __( 'Test', 'textdomain' ),
				'singular_name' => __( 'Test', 'textdomain' ),
				'add_new_item'  => __( 'Add New Test', 'textdomain' ),
			),
		)
	);
```

### Screenshots, screen recording, code snippet

# Note: This might be intermittent
## We have had some team members test this and have results both ways
<img width="350" alt="Screen Shot 2023-02-23 at 3 04 25 PM" src="https://user-images.githubusercontent.com/7538525/221022718-c60897cd-d462-4ab6-84c2-94cc1bd6b8fb.png">
<img width="790" alt="Screen Shot 2023-02-23 at 3 15 04 PM" src="https://user-images.githubusercontent.com/7538525/221022751-c43ae087-c69a-420b-9cea-97b30c62e216.png">

### Environment info

- WordPress 6.2-beta3
- Gutenberg disabled

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by andrewserong** [comment]
I'm wondering if this issue is similar to https://github.com/WordPress/gutenberg/issues/48355 in that it could be to do with the post editor being unable to retrieve a post content block to determine which layout/alignment rules to make available. There is a fix for that issue over in https://github.com/WordPress/gutenberg/pull/48386 (make sure the fallback layout is used) which could potentially fix this one, too.

**Comment by tellthemachines** [comment]
I tested on trunk with the reproduction steps above and was able to reproduce the issue. I can also confirm that it is now fixed by #48386. I'm going to go ahead and close this issue for now.
